packages/amplication-client/src/Profile/hooks/UseCurrentUser.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from "react";
+import * as models from "../../models";
+import { useQuery } from "@apollo/client";
+import useAuthenticated from "../../authentication/use-authenticated";
+import { GET_USER } from "../accountQueries";
+import { identity } from "../../util/analytics";
+
+type TData = {
+  me: {
+    account: models.Account;
+  };
+};
+
+export const useCurrentUser = () => {
```

Comment:
this is redundant. on every route we are using  useAuthenticated, so you can use it from there and also you need to remember that when you using a hook on a component, every time yo rernder it you'll restart your hook. so you better place it on a higher level or do it on a route level/App.
also, we are calling GET_USER at the higher level, so it will create another call for it